### PR TITLE
FE-970 - Hibana style cleanup for AB Testing pages

### PR DESCRIPTION
**Pull request by shrirupa**
FE-970 - Hibana style cleanup for AB Testing pages

### What Changed
- Updated the following routes for Hibana
 -> /ab-testing
 -> /ab-testing/create
 -> /ab-testing/:id/:version

- Added `<OGOnlyWrapper>` - this component selectively displays a wrapper only for OG version of app.

### How To Test
- Test the flow for Hibana and OG version for following routes - 
 -> /ab-testing
 -> /ab-testing/create
 -> /ab-testing/:id/:version

- Check if it similar to mocks -  https://sparkpost.invisionapp.com/share/D4WOUN5VUTE#/screens

### To Do
- [ ] Address Feedback

### CODE REVIEW

File: src/pages/abTesting/CreatePage.container.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (shrirupa) [inline src/pages/abTesting/CreatePage.container.js:1]: Created a separate container as shallow rendering and snapshots were becoming a problem using `<HibanaConsumer> `

File: src/pages/abTesting/CreatePage.js [hunk 2]
```diff
@@ -11,8 +11,7 @@ import { listTemplates } from 'src/actions/templates';
 // Components
 import { PageLink } from 'src/components/links';
 import { Panel, Page } from 'src/components/matchbox';
-import AbTestCreateForm from './components/AbTestCreateForm';
-
+import AbTestCreateForm from './components/AbTestCreateForm.container';
```
Comment (shrirupa) [inline src/pages/abTesting/CreatePage.js:14]: changed the import to be from container

File: src/pages/abTesting/CreatePageHibana.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    return createAbTestDraft({ abTest, subaccount: subaccountId }).then(() => {
+      showAlert({ type: 'success', message: 'A/B test draft created' });
+      history.push(`/ab-testing/${abTest.id}/1${setSubaccountQuery(subaccountId)}`);
+    });
+  };
+
+  render() {
+    return (
+      <Page
+        breadcrumbAction={{ content: 'Back to A/B Tests', component: PageLink, to: '/ab-testing' }}
```
Comment (shrirupa) [inline src/pages/abTesting/CreatePageHibana.js:37]: this changed from the OG component.

File: src/pages/abTesting/CreatePageHibana.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+// Actions
+import { createAbTestDraft } from 'src/actions/abTesting';
+import { showAlert } from 'src/actions/globalAlert';
+import { listTemplates } from 'src/actions/templates';
+
+// Components
+import { PageLink } from 'src/components/links';
+import { Panel, Page, Text } from 'src/components/matchbox';
+import AbTestCreateForm from './components/AbTestCreateForm.container';
```
Comment (shrirupa) [inline src/pages/abTesting/CreatePageHibana.js:13]: changed the import to be from container

File: src/pages/abTesting/CreatePageHibana.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  };
+
+  render() {
+    return (
+      <Page
+        breadcrumbAction={{ content: 'Back to A/B Tests', component: PageLink, to: '/ab-testing' }}
+      >
+        <Text as="h1" fontWeight="500" marginTop={-15} mb={20}>
+          Create a New A/B Test
+        </Text>
```
Comment (nicklemmon) [inline src/pages/abTesting/CreatePageHibana.js:41]: Should we just use the `title` prop within the `<Page/>` component? Looks like you sort of re-created it here.

File: src/pages/abTesting/CreatePageHibana.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    const { id, name, subaccount, default_variant = {} } = values;
+    const subaccountId = subaccount ? subaccount.id : null;
+    const default_template = { template_id: default_variant.id };
+    const abTest = { id, name, default_template };
+
+    return createAbTestDraft({ abTest, subaccount: subaccountId }).then(() => {
+      showAlert({ type: 'success', message: 'A/B test draft created' });
+      history.push(`/ab-testing/${abTest.id}/1${setSubaccountQuery(subaccountId)}`);
+    });
+  };
```
Comment (nicklemmon) [inline src/pages/abTesting/CreatePageHibana.js:32]: I'm a little worried about all of this duplicated logic and having two sets of UI. Could we just update the user experience in both versions of the app so you don't have to have two versions of each component?

Seems to look fine in the OG style!

Screen Shot 2020-04-23 at 12 30 14 PM

File: src/pages/abTesting/components/view/VariantsView.js [hunk 7]
```diff
@@ -71,31 +74,34 @@ export const Engagement = ({ variant, showRate }) => {
...
+            // eg { id: 'temp', shared_with_all: true } vs. { id: 'temp', subaccount_id: 101 }
+            //    both are usable from the same ab test
+            to: `/templates/edit/${variant.template_id}/published`,
+          },
+        ]}
+      >
+        {title && <h6 className={styles.SmallHeader}>{title}</h6>}
+        <LabelledValue label="Template ID">
+          <h6>{variant.template_id}</h6>
+        </LabelledValue>
```
Comment (nicklemmon) [inline src/pages/abTesting/components/view/VariantsView.js:98]: Screen Shot 2020-04-23 at 12 32 38 PM

Could use some space within these sections - the titles (in this case 'Default Template' and 'Variant 1') are too close to the form field labels below them.

File: src/pages/abTesting/components/view/ViewHibana.module.scss [hunk 8]
```diff
@@ -0,0 +1,9 @@
+@import '~@sparkpost/matchbox/src/styles/config.scss';
+
+.HelpText {
+  font-size: initial;
+}
+
+.SmallHeader {
+  font-size: initial;
+}
```
Comment (nicklemmon) [inline src/pages/abTesting/components/view/ViewHibana.module.scss:9]: You'll need to override the color for that `.InfoIcon` class as well using the updated design tokens

File: src/pages/abTesting/components/content/StatusContent.js [hunk 9]
```diff
@@ -1,71 +1,80 @@
...
 const StatusContent = ({ test, rescheduling }) => (
   <div>
     {(test.status === 'draft' || rescheduling) && (
       <div>
-        <p>
+        <ContentText fontSize="200" color="gray.600">
           You can update your Test Name and select a start time and end time to schedule your A/B
           Test.
-        </p>
+        </ContentText>
```
Comment (nicklemmon) [inline src/pages/abTesting/components/content/StatusContent.js:12]: Looks like there could be some space between these paragraphs. `<Stack/>` wrapping both should handle it:

Screen Shot 2020-04-23 at 12 36 38 PM

File: src/pages/abTesting/components/content/ContentText.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { useHibana } from 'src/context/HibanaContext';
+import { Text } from 'src/components/matchbox';
+
+const ContentText = props => {
+  const [state] = useHibana();
+  const { isHibanaEnabled } = state;
+  return isHibanaEnabled ? <Text {...props} /> : <p>{props.children}</p>;
+};
+
```
Comment (nicklemmon) [inline src/pages/abTesting/components/content/ContentText.js:10]: This is a good idea - we should potentially make this in to a global component, but not a priority now I don't think.

File: src/pages/abTesting/components/StatusPanel.js [hunk 11]
```diff
@@ -55,7 +54,7 @@ export const StatusPanel = ({ test, version, id, subaccountId, latest, subaccoun
   }
 
   return (
-    <Panel>
+    <OGOnlyWrapper as={Panel}>
```
Comment (nicklemmon) [inline src/pages/abTesting/components/StatusPanel.js:57]: Nice. This is very useful.

File: src/pages/abTesting/EditMode.js [hunk 12]
```diff
@@ -17,7 +17,7 @@ import { listTemplates } from 'src/actions/templates';
 import { selectEditInitialValues } from 'src/selectors/abTesting';
```
Comment (nicklemmon) [inline src/pages/abTesting/EditMode.js:17]: Screen Shot 2020-04-23 at 12 47 59 PM

The loader makes the delete modal look huge for Hibana - I would update it to use the `<PanelLoading/>` component and that should address it.

File: src/pages/abTesting/index.js [hunk 13]
```diff
@@ -1,9 +1,9 @@
 import ListPage from './ListPage';
 import DetailsPage from './DetailsPage';
-import CreatePage from './CreatePage';
+import CreatePage from './CreatePage.container';
 
 export default {
   ListPage,
```
Comment (nicklemmon) [inline src/pages/abTesting/index.js:6]: I don't see any mocks for the list page, but in the other list mocks I've seen, the links aren't bolded anymore. I've been able to get around this with:

```
<strong>
  <Text fontWeight="400">Click Me</Text>
</strong>
```

_Or_ your `OGOnlyWrapper` might do well here.

Comment (nicklemmon) [reply src/pages/abTesting/index.js:6]: The list page is also using some custom CSS that could use an override with the Hibana design tokens - just noticed that (`src/pages/abTesting/ListPage.module.scss`)

File: src/pages/abTesting/components/Section.js [hunk 14]
```diff
@@ -1,31 +1,53 @@
-import React, { Component } from 'react';
-import { Grid } from 'src/components/matchbox';
+import React from 'react';
+import { Box, Grid, OGOnlyWrapper, Text } from 'src/components/matchbox';
 import styles from './Section.module.scss';
```
Comment (nicklemmon) [inline src/pages/abTesting/components/Section.js:3]: Looks like this has some CSS that needs to be overridden by newer Matchbox tokens.

File: src/pages/abTesting/components/fields/SettingsFields.js [hunk 15]
```diff
@@ -1,88 +1,89 @@
...
-          label='How should we determine a winning variant?'
+          label="How should we determine a winning variant?"
           options={[
             { label: 'By Unique Click Rate', value: 'count_unique_clicked' },
-            { label: 'By Unique Open Rate', value: 'count_unique_confirmed_opened' }
+            { label: 'By Unique Open Rate', value: 'count_unique_confirmed_opened' },
           ]}
           disabled={disabled}
         />
         <Field
```
Comment (nicklemmon) [inline src/pages/abTesting/components/fields/SettingsFields.js:59]: Looks like there should be some space between these fields - `<Stack/>` should handle it.

File: src/pages/abTesting/CreatePageHibana.js (hunk 5 above)
Comment (shrirupa) [reply src/pages/abTesting/CreatePageHibana.js:41]: Thanks! I missed this prop.

File: src/pages/abTesting/CreatePageHibana.js (hunk 6 above)
Comment (shrirupa) [reply src/pages/abTesting/CreatePageHibana.js:32]: So the OG version will then have the missing link for `Learn more about A/B test`

**Comment by shrirupa** [comment]
@nicklemmon ready for another pass 🙂

File: src/components/panelLoading/PanelLoading.js [hunk 16]
```diff
@@ -5,10 +5,16 @@ import { Loading } from 'src/components/loading/Loading';
...
+  const { minHeight, accent, title } = props;
 
   return (
-    <Panel className={styles.Loading} style={{ minHeight }} accent={accent} data-id="panel-loading">
+    <Panel
+      className={styles.Loading}
+      style={{ minHeight }}
+      accent={accent}
+      data-id="panel-loading"
+      title={title}
```
Comment (nicklemmon) [inline src/components/panelLoading/PanelLoading.js:16]: I'm not sure why we didn't introduce this sooner 😨 - well done.

File: src/pages/abTesting/components/SectionHibana.module.scss [hunk 17]
```diff
@@ -0,0 +1,8 @@
+@import '~@sparkpost/design-tokens-hibana/tokens';
+
+.Right {
+  margin-top: spacing(200);
+  @media (min-width: media-query(md)) {
+    margin-top: rem(-25);
+  }
```
Comment (nicklemmon) [inline src/pages/abTesting/components/SectionHibana.module.scss:7]: I think the purpose of the negative margin is to line these up?

Screen Shot 2020-04-24 at 8 10 19 PM

File: src/pages/abTesting/CreatePage.js [hunk 18]
```diff
@@ -37,18 +35,9 @@ export class CreatePage extends Component {
     return (
       <Page
         breadcrumbAction={{ content: 'Back to A/B Tests', component: PageLink, to: '/ab-testing' }}
+        title="Create a New A/B Test"
```
Comment (nicklemmon) [inline src/pages/abTesting/CreatePage.js:38]: 👏

File: src/pages/abTesting/components/Section.js [hunk 19]
```diff
@@ -1,31 +1,58 @@
...
+  </OGOnlyWrapper>
 );
 
-export default class Section extends Component {
-  static Left = Left;
-  static Right = Right;
+const Right = ({ children }) => {
+  const styles = useHibanaOverride(OGStyles, HibanaStyles);
+  return (
+    <OGOnlyWrapper as={Grid.Column} xs={12} lg={7}>
```
Comment (nicklemmon) [inline src/pages/abTesting/components/Section.js:20]: This has turn out to be so helpful

File: src/pages/abTesting/components/Section.js [hunk 20]
```diff
@@ -1,31 +1,58 @@
...
+function HibanaSection(props) {
+  const { title, children } = props;
 
-  render() {
-    const { title, children } = this.props;
+  return (
+    <div>
+      <Text as="span" fontSize="400" fontWeight="600" role="heading" aria-level="3" mb="200">
+        {title}
+      </Text>
```
Comment (nicklemmon) [inline src/pages/abTesting/components/Section.js:46]: I think I'm going to make a `<Heading/>` component in another branch to help with this sort of thing, and basically copy this API:

https://twitter.com/wesbos/status/1247607715112923138

Probably this though:
```
<Heading as="h2" looksLike="h3"/>
```



For now this looks great.

File: src/pages/abTesting/components/SectionHibana.module.scss (hunk 17 above)
Comment (shrirupa) [reply src/pages/abTesting/components/SectionHibana.module.scss:7]: ohh... this doesn't seem to have a perfect value it needs to be different based on Status the ab test is in 😕 . I will try and see how to fix this.

**Pull request merged**